src/poetry/repositories/pool.py

```diff
@@ -62,134 +70,62 @@ def add_repository(
         """
         Adds a repository to the pool.
         """
-        repository_name = repository.name.lower()
-        if repository_name in self._lookup:
-            raise ValueError(f"{repository_name} already added")
+        if repository in self.repositories:
+            raise ValueError(f"{repository.name.lower()} already added")
 
-        if default:
-            if self.has_default():
-                raise ValueError("Only one repository can be the default")
-
-            self._default = True
-            self._repositories.insert(0, repository)
-            for name in self._lookup:
-                self._lookup[name] += 1
+        if default and self.has_default():
+            raise ValueError("Only one repository can be the default")
 
-            if self._secondary_start_idx is not None:
-                self._secondary_start_idx += 1
-
-            self._lookup[repository_name] = 0
+        if default:
+            repository_priority = RepositoryPriority.DEFAULT
         elif secondary:
-            if self._secondary_start_idx is None:
-                self._secondary_start_idx = len(self._repositories)
-
-            self._repositories.append(repository)
-            self._lookup[repository_name] = len(self._repositories) - 1
+            repository_priority = RepositoryPriority.SECONDARY
         else:
-            self._has_primary_repositories = True
-            if self._secondary_start_idx is None:
-                self._repositories.append(repository)
-                self._lookup[repository_name] = len(self._repositories) - 1
-            else:
-                self._repositories.insert(self._secondary_start_idx, repository)
-
-                for name, idx in self._lookup.items():
-                    if idx < self._secondary_start_idx:
-                        continue
-
-                    self._lookup[name] += 1
-
-                self._lookup[repository_name] = self._secondary_start_idx
-                self._secondary_start_idx += 1
-
+            repository_priority = RepositoryPriority.PRIMARY
+        self._repositories[repository] = repository_priority
         return self
 
     def remove_repository(self, repository_name: str) -> Pool:
-        if repository_name is not None:
-            repository_name = repository_name.lower()
-
-        idx = self._lookup.get(repository_name)
-        if idx is not None:
-            del self._repositories[idx]
-            del self._lookup[repository_name]
-
-            if idx == 0:
-                self._default = False
-
-            for name in self._lookup:
-                if self._lookup[name] > idx:
-                    self._lookup[name] -= 1
-
-            if (
-                self._secondary_start_idx is not None
-                and self._secondary_start_idx > idx
-            ):
-                self._secondary_start_idx -= 1
-
+        for repo in self.repositories:
+            if repo.name.lower() == repository_name.lower():
```

Comment:
While we are refactoring, I think we should hoist any normalization up and out (or make it explicit that names are not normalized) -- having this only in the remove code is surprising.

Reply:
relevant: https://github.com/python-poetry/poetry/issues/6670#issuecomment-1264443658

Reply:
Given that comment, I think we should normalize names uniformly.

Reply:
Done.